addons/a11y/src/components/Report/Rules.tsx

```diff
@@ -40,30 +53,63 @@ const Status = styled.div(({ passes, impact }: { passes: boolean; impact: string
   },
 }));
 
+export enum ImpactValue {
+  MINOR = 'minor',
+  MODERATE = 'moderate',
+  SERIOUS = 'serious',
+  CRITICAL = 'critical',
+}
+
 interface RuleProps {
   rule: CheckResult;
-  passes: boolean;
 }
 
-const Rule: FunctionComponent<RuleProps> = ({ rule, passes }) => (
-  <Item>
-    <Status passes={passes || undefined} impact={rule.impact}>
-      {passes ? <Icons icon="check" /> : <Icons icon="cross" />}
-    </Status>
-    <Message>{rule.message}</Message>
-  </Item>
-);
+const formatSeverityText = (severity: string) => {
+  return severity
+    .charAt(0)
+    .toUpperCase()
+    .concat(severity.slice(1));
+};
+
+const Rule: FunctionComponent<RuleProps> = ({ rule }) => {
+  let badgeType = '';
+  switch (rule.impact) {
+    case ImpactValue.CRITICAL:
+      badgeType = 'critical';
+      break;
+    case ImpactValue.SERIOUS:
+      badgeType = 'negative';
+      break;
+    case ImpactValue.MODERATE:
+      badgeType = 'warning';
+      break;
+    case ImpactValue.MINOR:
+      badgeType = 'neutral';
+      break;
+    default:
+      break;
+  }
```

Comment:
```suggestion
const badgeType = rule.impact in ImpactValue ? rule.impact : null;
```
can simplify?

Reply:
I think that would be a good solution if the badge names were the same as the severity names but unfortunately, they are not

Reply:
I’ve looked into this a bit more and the best option as far as performance goes is to use a switch statement. So I’d like to leave it as is.

This article goes into some of the pros and cons which I found interesting: https://medium.com/front-end-weekly/switch-case-if-else-or-a-lookup-map-a-study-case-de1c801d944